ecosystem/typescript/sdk_v2/src/crypto/authentication_key.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { sha3_256 as sha3Hash } from "@noble/hashes/sha3";
+import { AccountAddress, Hex } from "../core";
+import { HexInput } from "../types";
+import { MultiEd25519PublicKey } from "./multi_ed25519";
+import { PublicKey } from "./asymmetric_crypto";
+
+/**
+ * Each account stores an authentication key. Authentication key enables account owners to rotate
+ * their private key(s) associated with the account without changing the address that hosts their account.
+ * @see {@link https://aptos.dev/concepts/accounts | Account Basics}
+ * 
+ * Note: AuthenticationKey only supports Ed25519 and MultiEd25519 public keys for now.
+ *
+ * Account addresses can be derived from AuthenticationKey
+ */
+export class AuthenticationKey {
+  // Length of AuthenticationKey in bytes(Uint8Array)
+  static readonly LENGTH: number = 32;
+
+  // Scheme identifier for MultiEd25519 signatures used to derive authentication keys for MultiEd25519 public keys
+  static readonly MULTI_ED25519_SCHEME: number = 1;
+
+  // Scheme identifier for Ed25519 signatures used to derive authentication key for MultiEd25519 public key
+  static readonly ED25519_SCHEME: number = 0;
```

Comment:
Note that a full list of schemes is here: https://github.com/aptos-labs/aptos-core/blob/c2c8ce30f39c6bf1e69b4db01d47c64e6c9a3cfd/types/src/transaction/authenticator.rs#L349-L365

I'd probably put them in numerical order

Reply:
I will add a scheme enum to TS SDK.